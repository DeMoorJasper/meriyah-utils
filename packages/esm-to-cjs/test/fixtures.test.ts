import fs from "fs";
import path from "path";
import * as globby from "globby";
import { convertEsModule } from "./test-utils";

describe("Fixtures", () => {
  const FIXTURES_DIR = path.join(__dirname, "fixtures");
  const fixtures = globby.sync("**.js", {
    cwd: FIXTURES_DIR,
  });

  for (let fixture of fixtures) {
    it(`fixture: ${fixture}`, () => {
      const filepath = path.join(FIXTURES_DIR, fixture);
      const code = fs.readFileSync(filepath, "utf-8");
      const result = convertEsModule(code);
      expect(result).toMatchSnapshot();
    });
  }
});
