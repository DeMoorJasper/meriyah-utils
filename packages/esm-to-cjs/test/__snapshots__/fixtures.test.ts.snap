// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixtures fixture: class-properties.js 1`] = `
"exports.LocalFontSource = void 0;
Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
var $csb__react = require(\\"react\\");
var React__default = $_csb__interopRequireDefault($csb__react);
var $csb___fonts_js = require(\\"./fonts.js\\");
class LocalFontSource {
  React_default;

  typefaceAliases;

  constructor() {
    this.typefaceAliases = new Map();
    this.React_default = \\"react\\";
  }
  test() {
    this.React_default = \\"another-react\\";
    (0, $csb___fonts_js.typefaceAliases)();
    return this.typefaceAliases;
  }
}
exports.LocalFontSource = LocalFontSource;
function $_csb__interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : {
    default: obj
  };
}
"
`;

exports[`Fixtures fixture: function-args.js 1`] = `
"var $csb__lodash = require(\\"lodash\\");
var Lodash = $csb__lodash;
function doSomething(Lodash) {
  return Lodash.flatten([]);
}
"
`;

exports[`Fixtures fixture: scoped-exports.js 1`] = `
"Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
var $csb__react = require(\\"react\\");
var __$csb_exports = [];
function test(__$csb_exports) {
  return __$csb_exports;
}
__$csb_exports.push(\\"test\\");
Object.keys($csb__react).forEach(function (key) {
  if (key === \\"default\\" || key === \\"__esModule\\") return;
  if (Object.prototype.hasOwnProperty.call(exports, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    configurable: true,
    get: function $csbGet() {
      return $csb__react[key];
    }
  });
});
"
`;
